### PR TITLE
changes to build to include skydive cleanup and debug of skydive

### PR DESCRIPTION
**Pull request by hunchback**
The scope of these changes:
- cleanup - added `make skydive.cleanup` (and some other targets) to enable deleting artifacts created when building `make skydive`
- refactoring - mainly broke very long single lines within Makefile to multiline
- debug - added helper targets to build `make debug`, and run `make debug.agent` / `make debug.analyzer`
- go installation - added `make golang` to automate installation of Go

### CODE REVIEW

**Comment by hunchback** [comment]
The build server is out of space:

```
7:03:30.587 compile: writing output: write $WORK/github.com/skydive-project/skydive/topology/graph/_test/github.com/skydive-project/skydive/topology/graph.a: no space left on device
17:03:30.645 # github.com/skydive-project/skydive/vendor/github.com/nlewo/contrail-introspect-cli/descriptions
17:03:30.645 compile: writing output: write $WORK/github.com/skydive-project/skydive/vendor/github.com/nlewo/contrail-introspect-cli/descriptions.a: no space left on device
17:03:30.866 # github.com/skydive-project/skydive/topology/graph/traversal
17:03:30.866 compile: writing output: write $WORK/github.com/skydive-project/skydive/topology/graph/traversal.a: no space left on device
17:03:30.998 # github.com/skydive-project/skydive/topology/graph/traversal
17:03:30.998 compile: writing output: write $WORK/github.com/skydive-project/skydive/topology/graph/traversal/_test/github.com/skydive-project/skydive/topology/graph/traversal.a: **no space left on device**
```

**Comment by safchain** [comment]
rerun tests

**Comment by safchain** [comment]
fixed the disk space of the 3rd build server

File: Makefile [hunk 1]
```diff
@@ -63,6 +63,29 @@ endif
 .PHONY: all doc
 all: install
 
+.PHONY: dlv
+dlv:
+	go get github.com/derekparker/delve/cmd/dlv
```
Comment (safchain) [inline Makefile]: I would prefer to not install/target dev tool as it is not the purpose of this Makefile. That said I'm ok with debug target to add flags.

File: Makefile [hunk 2]
```diff
@@ -47,6 +47,18 @@ endif
...
 all: install
 
+GO_DOWNLOAD:=https://storage.googleapis.com/golang
+GO_VERSION:=1.9.2
+GO_OS:=linux
+GO_ARCH:=amd64
+GO_PKG:=go$(GO_VERSION).$(GO_OS)-$(GO_ARCH).tar.gz
+
+.PHONY: golang
+golang:
```
Comment (safchain) [inline Makefile]: I would prefer to add golang installation things. This is not the goal of Skydive to install proper Go environment. I know that lot of us have different GOROOT environment.

Comment (hunchback) [reply Makefile]: Will remove `golang` target

**Comment by hunchback** [comment]
Failed build due to what appears unrelated error:

```
12:14:07.775 --- FAIL: TestFlowsEBPF (1.34s)
12:14:07.775 	tests.go:339: Failed to setup test: Failed to create injectpacket, 400 Bad Request: Not able to find a dest node and dest IP also empty, graph: 
12:14:07.775 	
```

**Comment by safchain** [comment]
@hunchback going to rerun it. Can you please have a look at the remaining comment.

**Comment by safchain** [comment]
rerun functional-tests-backend-elasticsearch

File: Makefile (hunk 1 above)
Comment (hunchback) [reply Makefile]: @safchain - removed `dlv` target

**Comment by hunchback** [comment]
@safchain - any additional changes needed here?

**Pull request merged**